src/modules/findProvinceFromCoordinate/index.ts

```diff
@@ -0,0 +1,74 @@
+import { GeoJSONData } from "./IRGeoJSON";
+
+interface Point {
+	longitude: number;
+	latitude: number;
+}
+
+interface GeoJSONFeature {
+	properties: Record<string, string | undefined>;
+	geometry: {
+		type: string;
+		coordinates: object; // This represents MultiPolygon coordinates
+	};
+}
+
+interface Province {
+	properties: {
+		'name:fa': string;
+		'name:en': string;
+	};
+	geometry: object; 
+}
+
+const provinces = GeoJSONData;
+
+function pointInPolygon(polygon: number[][], point: Point): boolean {
+	let odd = false;
+	for (let i = 0, j = polygon.length - 1; i < polygon.length; i++) {
+		if (((polygon[i][1] > point.latitude) !== (polygon[j][1] > point.latitude))
+			&& (point.longitude < ((polygon[j][0] - polygon[i][0]) * (point.latitude - polygon[i][1]) / (polygon[j][1] - polygon[i][1]) + polygon[i][0]))) {
+			odd = !odd;
+		}
+		j = i;
+	}
+	return odd;
+}
+
+/**
+ * Find the province from a given coordinate point.
+ * @param pointToCheck The coordinate point to check.
+ * @returns The province information.
+ * @throws {Error} If the province cannot be found based on the provided coordinates.
+ * @example
+ * const point: Point = { latitude: 35.6892, longitude: 51.3890 };
+ * const province = findProvinceFromCoordinate(point);
+ * console.log(province.name); // "تهران"
+ * console.log(province.properties['name:fa']); // "تهران"
+ * console.log(province.properties['name:en']); // "Tehran"
+ */
+export const findProvinceFromCoordinate = (pointToCheck: Point): Province => {
+	let foundProvince: GeoJSONFeature | undefined;
+	for (let index = 0; index < provinces.features.length; index++) {
+		const province = provinces.features[index];
+		const isInsideProvince = pointInPolygon(province.geometry.coordinates[0][0], pointToCheck);
+		if (isInsideProvince) {
+			foundProvince = province;
+			break;
+		}
+	}
+	if (foundProvince) {
+		const normalizedProvinceObject: Province = {
+			properties: {
+				'name:fa': foundProvince.properties.name || "",
+				'name:en': foundProvince.properties['name:en'] || "",
+			},
+			geometry: foundProvince.geometry
```

Comment:
why are you  having keys like `name:fa` ? 
it would not be clean to destruct this object and get the result 

i would suggest change to something like 

```js
const normalizedProvinceObject: Province = {
			name: {
				fa: foundProvince.properties.name || "",
			  en: foundProvince.properties['name:en'] || "",
			},
			geometry: foundProvince.geometry
}

// accessing
// .1
const {fa, en} = province.name

// .2
const { name } = province
name.fa // تهران
name.en // tehran



```